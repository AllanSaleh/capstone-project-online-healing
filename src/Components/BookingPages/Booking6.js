import React from 'react';

export default function Booking4() {
  return (
    <div className="flex flex-col justify-evenly h-firstsection px-sides pt-navbar">
      <div>
        <h1 className="text-title uppercase mb-4">What brings you here?</h1>
        <h3 className="text-subtitle opacity-50 text-justify">
          Please specify (in a few sentences) why you'd like counseling.
          <br />
          This will give your counselor a good understanding of where to start.
        </h3>
      </div>

      <textarea
        placeholder="Please explain here..."
        className="text-paragraph self-center w-2/3 flex flex-col justify-evenly h-1/2 p-8 shadow-lg"
      />
      <div className="mx-auto w-1/2 flex justify-between">
        <button
          type="button"
          className="uppercase bg-blue-dark h-12 w-32 text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
        >
          back
        </button>
        <button
          type="button"
          className="uppercase bg-blue-dark h-12 w-32 text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
        >
          next
        </button>
      </div>
    </div>
  );
}
