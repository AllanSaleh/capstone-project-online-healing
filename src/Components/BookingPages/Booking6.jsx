import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';

export default function Booking6() {
  window.scrollTo(0, 0);
  const { t } = useTranslation();

  const question = t('Booking.Title2');

  const history = useHistory();
  const [userAnswer, setUserAnswer] = useState('');
  console.log(userAnswer);

  const booking = JSON.parse(localStorage.getItem('userBooking'));
  const PrevPage = () => {
    booking.choices.pop();
    localStorage.setItem('userBooking', JSON.stringify(booking));
    history.push('/Booking5');
  };
  const NextPage = () => {
    if (!userAnswer) {
      alert('please provide an answer');
      return;
    }
    const choice = {
      question,
      answer: userAnswer,
    };
    booking.choices.push(choice);
    localStorage.setItem('userBooking', JSON.stringify(booking));
    history.push('/Booking7');
  };
  return (
    <div className="flex flex-col justify-evenly lg:h-firstsection px-sides pt-navbar">
      <div>
        <h1 className="text-3xl lg:text-title uppercase my-8">{question}</h1>
        <h3 className="text-md lg:text-subtitle opacity-50 text-justify">
          {t('Booking.Subtitle2')}
        </h3>
      </div>

      <textarea
        placeholder={t('Booking.Explain')}
        className="text-md lg:text-paragraph outline-none resize-none w-full lg:w-4/5 lg:mx-auto flex flex-col justify-evenly h-1/2 p-8 shadow-lg"
        onChange={(event) => {
          setUserAnswer(event.target.value);
        }}
      />
      <div className="mx-auto w-full lg:w-1/2 flex justify-between mt-8 mb-4">
        <button
          onClick={() => PrevPage()}
          type="button"
          className="uppercase bg-blue-dark h-12 w-32 text-md lg:text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
        >
          {t('Booking.Back')}
        </button>
        <button
          onClick={() => NextPage()}
          type="button"
          className="uppercase bg-blue-dark h-12 w-32 text-md lg:text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition-all duration-300"
        >
          {t('Booking.Next')}
        </button>
      </div>
    </div>
  );
}
