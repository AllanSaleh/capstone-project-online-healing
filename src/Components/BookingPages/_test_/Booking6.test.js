import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Booking6 from '../Booking6';

it('Booking6 renders correctly', () => {
  const tree = renderer.create(<Booking6 />).toJSON();
  expect(tree).toMatchSnapshot();
});
