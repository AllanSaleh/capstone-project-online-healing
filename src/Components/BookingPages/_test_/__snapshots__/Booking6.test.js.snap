// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking6 renders correctly 1`] = `
<div
  className="flex flex-col justify-evenly h-firstsection px-sides pt-navbar"
>
  <div>
    <h1
      className="text-3xl lg:text-title uppercase my-8"
    >
      What brings you here?
    </h1>
    <h3
      className="text-md lg:text-subtitle opacity-50 text-justify"
    >
      Please specify (in a few sentences) why you'd like counseling.
      <br />
      This will give your counselor a good understanding of where to start.
    </h3>
  </div>
  <textarea
    className="text-md lg:text-paragraph outline-none resize-none w-full lg:w-4/5 lg:mx-auto flex flex-col justify-evenly h-1/2 p-8 shadow-lg"
    placeholder="Please explain here..."
  />
  <div
    className="mx-auto w-full lg:w-1/2 flex justify-between mt-8"
  >
    <button
      className="uppercase bg-blue-dark h-12 w-32 text-md lg:text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
      onClick={[Function]}
      type="button"
    >
      back
    </button>
    <button
      className="uppercase bg-blue-dark h-12 w-32 text-md lg:text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition-all duration-300"
      onClick={[Function]}
      type="button"
    >
      next
    </button>
  </div>
</div>
`;
